engine/src/gui/text_area.cpp

```diff
@@ -208,85 +204,80 @@ void TextArea::RenderTextItem(TextAreaItem *current, int level) {
         if (count < top_item_number) {
             count++;
         } else {
-            new_y = ycoord[5] - (text_spacing * (count + 1 - top_item_number)) + horizontal_spacer;
-            new_x = xcoord[5] + (horizontal_per_level * (level - 1));
+            float new_x = 0;
+            float new_y = 0;
+            new_y       = ycoord[5] - (text_spacing * (count + 1 - top_item_number)) + horizontal_spacer;
+            new_x       = xcoord[5] + (horizontal_per_level * (level - 1));
             GFXColorf(current->col);
             ShowText(new_x, new_y, width[5], font_size, current->description, do_multiline);
             count++;
         }
     }
-    max = current->child_count;
-    if (max <= 0) {
+    if (current->child_count() <= 0) {
         return;
     }
-    for (cur = 0; cur < max; cur++) {
-        RenderTextItem(current->child[cur], level + 1);
+    for (const auto& current_child : current->children) {
+        RenderTextItem(current_child, level + 1);
     }
 }
 
 void TextArea::AddTextItem(const char *name, const char *description, const char *parent_name, const GFXColor col) {
-    TextAreaItem *master;
-    master = ItemList->FindChild(parent_name);
-    item_count++;
-    if (master == NULL) {
-        ItemList->AddChild(name, description, col);
-    } else {
+    const SharedPtr<TextAreaItem> master = item_tree_top->FindChild(parent_name);
+    ++item_count;
+    if (master) {
         master->AddChild(name, description, col);
+    } else {
+        item_tree_top->AddChild(name, description, col);
     }
 }
 
 void TextArea::ChangeTextItem(const char *name, const char *description, bool wrap) {
-    TextAreaItem *search;
-    search = ItemList->FindChild(name);
-    if (search == 0) {
+    const SharedPtr<TextAreaItem> search = item_tree_top->FindChild(name);
+    if (!search) {
         return;
     }
-    if (search->description != 0) {
+    if (search->description != nullptr) {
         free(search->description);
+        search->description = nullptr;
     }
-    search->description = strdup(description);
+    search->description = vega_str_dup(description);
```

Comment:
Definitely C. Why would anyone write char* when std::string is available. I wonder how hard it would be to replace` char*` with `std::string`?

Reply:
> Definitely C. Why would anyone write char* when std::string is available. I wonder how hard it would be to replace` char*` with `std::string`?

For the most part it's easy unless you're doing buffer allocation - then it can be easier to allocate it using `char*` and then convert after.